### PR TITLE
add a second creatable option

### PR DESCRIPTION
**Pull request by ryanabooth**
Adding a second newOptionCreator so that we can pass in two different options when a user types something unique.

`src/Creatable.js` is the precompliation source file that contains the main changes.
`lib/Creatable.js` is the compiled version
`examples/src/components/Creatable.js` is the example on the doc page that's useful for testing / demoing the change.

The rest of the files were compiled by running `npm run build`.

https://app.asana.com/0/281605278266465/374433993661621/f

### CODE REVIEW

**Review by vladtsf — APPROVED** [review]
image

File: dist/react-select.js [hunk 1]
```diff
@@ -440,16 +442,20 @@ var Creatable = (0, _createReactClass2['default'])({
 	},
 
 	createNewOption: function createNewOption() {
+		var useSecondaryCreator = arguments.length <= 0 || arguments[0] === undefined ? false : arguments[0];
```
Comment (jmondo) [inline dist/react-select.js:445]: not following

File: dist/react-select.js [hunk 2]
```diff
@@ -440,16 +442,20 @@ var Creatable = (0, _createReactClass2['default'])({
...
 		var _props = this.props;
 		var isValidNewOption = _props.isValidNewOption;
 		var newOptionCreator = _props.newOptionCreator;
+		var newOptionCreator2 = _props.newOptionCreator2;
 		var onNewOptionClick = _props.onNewOptionClick;
 		var _props$options = _props.options;
 		var options = _props$options === undefined ? [] : _props$options;
 		var shouldKeyDownEventCreateNewOption = _props.shouldKeyDownEventCreateNewOption;
 
+		var creator = useSecondaryCreator ? newOptionCreator2 : newOptionCreator;
```
Comment (jmondo) [inline dist/react-select.js:455]: should be consistent. is it secondary or 2

File: dist/react-select.js [hunk 3]
```diff
@@ -578,11 +612,11 @@ var Creatable = (0, _createReactClass2['default'])({
 	render: function render() {
 		var _this = this;
 
-		var _props4 = this.props;
-		var newOptionCreator = _props4.newOptionCreator;
-		var shouldKeyDownEventCreateNewOption = _props4.shouldKeyDownEventCreateNewOption;
+		var _props5 = this.props;
```
Comment (jmondo) [inline dist/react-select.js:615]: omg this lib is out of control

**Review by jmondo — COMMENTED** [review]
gonna need to go over this together i'm super confused

**Comment by ryanabooth** [comment]
@jmondo you reviewed the compiled code

File: examples/dist/bundle.js [hunk 4]
```diff
@@ -542,14 +576,14 @@ var Creatable = (0, _createReactClass2['default'])({
...
-		var onInputKeyDown = _props3.onInputKeyDown;
+		var _props4 = this.props;
+		var shouldKeyDownEventCreateNewOption = _props4.shouldKeyDownEventCreateNewOption;
+		var onInputKeyDown = _props4.onInputKeyDown;
 
 		var focusedOption = this.select.getFocusedOption();
 
-		if (focusedOption && focusedOption === this._createPlaceholderOption && shouldKeyDownEventCreateNewOption({ keyCode: event.keyCode })) {
-			this.createNewOption();
+		if (focusedOption && (focusedOption === this._createPlaceholderOption || this._createPlaceholderOption2 && focusedOption === this._createPlaceholderOption2) && shouldKeyDownEventCreateNewOption({ keyCode: event.keyCode })) {
```
Comment (adeeb1) [inline examples/dist/bundle.js:585]: Wrap the new logical operators in parenthesis so your intention is clear.

File: examples/dist/bundle.js [hunk 5]
```diff
@@ -559,8 +593,8 @@ var Creatable = (0, _createReactClass2['default'])({
 	},
 
 	onOptionSelect: function onOptionSelect(option, event) {
-		if (option === this._createPlaceholderOption) {
-			this.createNewOption();
+		if (option === this._createPlaceholderOption || this._createPlaceholderOption2 && option === this._createPlaceholderOption2) {
```
Comment (adeeb1) [inline examples/dist/bundle.js:596]: Same here with the logical operators.

File: examples/dist/standalone.js [hunk 6]
```diff
@@ -547,14 +581,14 @@ var Creatable = (0, _createReactClass2['default'])({
...
-		var onInputKeyDown = _props3.onInputKeyDown;
+		var _props4 = this.props;
+		var shouldKeyDownEventCreateNewOption = _props4.shouldKeyDownEventCreateNewOption;
+		var onInputKeyDown = _props4.onInputKeyDown;
 
 		var focusedOption = this.select.getFocusedOption();
 
-		if (focusedOption && focusedOption === this._createPlaceholderOption && shouldKeyDownEventCreateNewOption({ keyCode: event.keyCode })) {
-			this.createNewOption();
+		if (focusedOption && (focusedOption === this._createPlaceholderOption || this._createPlaceholderOption2 && focusedOption === this._createPlaceholderOption2) && shouldKeyDownEventCreateNewOption({ keyCode: event.keyCode })) {
```
Comment (adeeb1) [inline examples/dist/standalone.js]: Same here with the logical operators.

**Review by adeeb1 — CHANGES_REQUESTED** [review]
Looks good aside from some logical operators in all lines similar to this:

`(option === this._createPlaceholderOption || this._createPlaceholderOption2 && option === this._createPlaceholderOption2)`

Wrap in parenthesis the logical operators that should be evaluated together.

**Review by adeeb1 — APPROVED** [review]
Looks good! I unknowingly commented on the generated files, not the source files.

**Pull request merged**